rtmp-client/src/main/cpp/librtmp/rtmp.c

```diff
@@ -4971,16 +4971,17 @@ int
 RTMP_Read(RTMP *r, char *buf, int size)
 {
   int nRead = 0, total = 0;
+  int8_t readStatus;
 
   /* can't continue */
 fail:
-  switch (r->m_read.status) {
-  case RTMP_READ_EOF:
-  case RTMP_READ_COMPLETE:
-    return 0;
+  readStatus = r->m_read.status;
+  switch (readStatus) {
```

Comment:
Should still IMO explicitly `return readStatus` from the first case, so that it doesn't look like someone just forgot to put a return statement in there.

Reply:
Sorry about that, I've added it again.